src/decode-udp.c

```diff
@@ -56,16 +56,16 @@ static int DecodeUDPPacket(ThreadVars *t, Packet *p, const uint8_t *pkt, uint16_
         return -1;
     }
 
-    if (unlikely(len != UDP_GET_LEN(p))) {
-        ENGINE_SET_INVALID_EVENT(p, UDP_HLEN_INVALID);
+    if (unlikely(UDP_GET_LEN(p) < UDP_HEADER_LEN)) {
+        ENGINE_SET_INVALID_EVENT(p, UDP_PKT_TOO_SMALL);
```

Comment:
the issue here is that the length field isn't correct, not that the packet is too small?

Reply:
Yes. HLEN_INVALID also did not seem to fit.. I think in my head, I went with this bc the length received from the pkt is smaller than 8bytes..
New error code?

Reply:
The pkt too small class of events is meant to indicate the input data is too small, not that a length field is wrong

Reply:
Indeed. That was an incorrect event. So is UDP_HLEN_INVALID and UDP_HLEN_TOO_SMALL, would you agree? I'm thinking these indicate that the packet came with an invalid or too small header length. This event is however about the entire length of the packet.

Should we make a new event for this case?

Reply:
yeah I think so, something like `UDP_LEN_INVALID` or something